Task 031,Task 027, Task 028, Task 029, Task 030, and Task 033

mbagwell opened the pull request:
This commit gives the illusion that time is passing by making the sky and clouds fade and reappear over time.

mbagwell committed:
Updating Parallax Background

Updating the code for the background to accept the Sky.gd and Cloud.gd file I will be adding in

mbagwell committed:
Cloud Effect

Adding effect to clouds so that they fade and reappear over time.

mbagwell committed:
Sky Effects

creating a sky sprite/file so that the sky's background image will fade and reappear over time(right now time passes by quickly)

WilltheWys reviewed (commented):
Adding these changes makes the background look a little more pleasing. Making the Background Dynamic and changing is more pleasing to the eyes. It makes everything look like its moving.

NotHacker101 reviewed (approved):
I like that you explain what you did in a short way !!! keep up the good work.

mbagwell committed:
Add files via upload

Attempting to place the new bullet with the old bullet, since it is transparent around the edges of the box to look like a circle.

mbagwell committed:
Update Bullet.tscn

Sets the bullet's color, and decreases it's opacity so it is a brighter red

mbagwell committed:
Bullet Gravity

sets the bullet's gravity to zero so it shoots in a straight line continuously.

mbagwell committed:
Hover function Improvement

Allows the user to constantly hover by HOLDING "H" down instead of constantly having to press it.

DSF256 commented:
This looks pretty cool; a day and night transition makes the game feel a lot more lively.  I would probably recommend making the day/night cycle transitions somewhat longer, though.  The sky changes every 1 second, which makes days go by a bit too quickly.  I would probably change it to once every 5 or 6 seconds.

DSF256 reviewed (changes requested):
Okay, I think I screwed that last one up.  Anyway, I think this is good save for the interval between sky changes.  I tested a 5-second interval in my own project, and it seemed to feel more natural.  Hope this helps.

reid-wynkoop commented on 2d/platformer/src/Level/Sky.gd:
```diff
@@ -0,0 +1,30 @@
+# Declare member variables here. Examples:
+# var a = 2
+# var b = "text"
+onready var sprite = $Sprite
```
does sprite ever get used?

mbagwell replied:
No.

mbagwell committed:
Reduce hover velocity to a straight line

mbagwell committed:
Adds automatic shooting go player's gun(4 shots per sec)

mbagwell committed:
Makes enemies move and jump at random speeds/times

mbagwell committed:
Reduces rate the clouds dims/shines(5s) and gives delay in between(20s)

mbagwell committed:
Reduces rate the sky dims/shines(5s) and gives delay in between(20s)
